Trees/DepthFirstSearch.js

```diff
@@ -4,35 +4,32 @@
  * DFS Algorithm for traversing or searching graph data structures.
 */
 
-function traverseDFS (root) {
-  const stack = [root]
+// traverses a give tree from specified root
+function traverseDFS (tree, root) {
+  const stack = []
   const res = []
-
+  stack.push(searchDFS(tree, root))
```

Comment:
Why are you calling `searchDFS` here?

Reply:
To find the node with given value. The root just has the value property, so to get the entire node with that value, searchDFS is called.

Reply:
hmm, I think the root should be given by "reference" (i.e. index or node object) rather than by value, which requires searching for it; either way it's not clear that `root` is actually the *value* of the root

Reply:
made changes in function parameter and variable naming to make it clear.